convert2rhel/pkghandler.py

```diff
@@ -517,13 +516,15 @@ def remove_excluded_pkgs():
 
 def remove_repofile_pkgs():
     """Remove those non-RHEL packages that contain YUM/DNF repofiles (/etc/yum.repos.d/*.repo) or affect variables
-    in the repofiles (e.g. $releasever)
+    in the repofiles (e.g. $releasever).
 
     We can't be removing them together with the other excluded packages - it wouldn't
     be possible to access and install subscription-manager dependencies. At the same time
     we can't install subscription-manager before removing the excluded packages
     as there would be a conflict with one of the excluded packages (rhn-client-tools).
     """
+    # The above comment is very confusing. What does it mean to not remove the packages with the other excluded packages? 
+    # It's not clear when the above comment is referring to the excluded packages and when it's referring to nonexcluded packages that need to be removed.
```

Comment:
Michal wrote this when he last updated this docstring:

commit b59a7e5fbf9c355487d3026ae78451109c3c64e2
Author: Michal Bocek <mbocek@redhat.com>
Date:   Fri Jan 29 19:24:42 2021 +0100

    Remove repofile pkgs after submgr installation
    
    -> We need the original system repositories available in order to have
    access to the dependencies of subscription-manager when we get to
    installing it with yum/dnf. For that reason we need to make sure the
    packages containg the system yum/dnf repofiles are removed only after
    installing subscription-manager.
    
    Remove release pkgs only after installing submgr and subscribing the
    system -> we were removing the release pkgs before installing the
    subscription-manager pkgs and we were passing the --releasever option to
    yum with a value for RHEL repos, not the original system repos. That
    caused the submgr installation to fail. And on CentOS 8, subscription
    of the system fails if there's no release package installed
    (centos-linux-release, redhat-release).

I'm going to take a guess at what this means but also going to CC @bocekm to erify that I am correct:

When convert2rhel runs, it removes some excluded packages (what precisely are excluded packages?).  Some of those packages contain non-RHEL repo files (files which tell yum and dnf where to look for packages to install on the system).  In order to install subscription-manager, we need those repo files to be present so that yum or dnf can find the packages that subscription-manager depends on.  So we have to remove these particular packages after subscription-manager is installed.

However, the rhn-client-tools excluded package must be removed before we install subscription-manager because the two packages conflict.

To meet both requirements, we first remove the excluded packages which do not contain repo files.  Then we install subscription-manager.  We then subscribe the system (due to the way subscribing systems works in CentOS 8).  Only after those three steps do we call this function which removes the excluded packages which do contain repo files.

@bocekm, did I get that right?

Reply:
You got it right, @abadger.

The [summary of the issue](https://github.com/oamg/convert2rhel/pull/107#issuecomment-726353163) I fixed by introducing the `remove_repofile_pkgs()` function:
`The problem is that we first uninstall the excluded pkgs like oraclelinux-release or centos-release, which provide the standard system repofiles. So after we uninstall them and we get to install subscription-manager, yum has no repos to get the subscription-manager deps from.`

BTW, here's an example of list of excluded pkgs we're removing at the start of the conversion: https://github.com/oamg/convert2rhel/blob/main/convert2rhel/data/8/x86_64/configs/centos-8-x86_64.cfg#L10. The excluded pkgs are those that either cause transaction failures or are vendor-specific (like logos).

Reply:
Thanks, updated the message to make it more clear which packages are being referred to. I do wonder how much info does the customer need - do we need to explain why exactly the non-RHEL packages in question weren't removed earlier?

Reply:
Thanks, it looks good. I don't think we necessarily need to tell them more than what we do now. But feel free to suggest change to any message printed to the user.

By the way, most of your changes in this PR are targeting strings in triple apostrophes `"""` which are not being printed to the user. These are just documentation strings for the code developers. Feel free to update these as well. Tthough it won't have an effect to what the user sees in the terminal when executing convert2rhel. The messages printed to the user are in `logger.` or `loggerinst.<loglevel>("<message>")`.

Reply:
@bocekm Oh, good to know... That would have saved me some time.